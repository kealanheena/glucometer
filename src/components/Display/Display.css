.display-container {
  align-items: center;
  background: var(--display-background-color);
  display: flex;
  height: var(--display-height);
  padding: 0 4%;
  margin: 4%;
  width: var(--display-width);
}

.display-value {
  color: var(--display-text-color);
  font-size: var(--display-text-size);
  font-family: var(--main-font);
  font-weight: var(--display-text-weight);
  text-align: center;
  overflow: hidden;
}

.unit-display {
  font-size: 0.4em;
}

.unit-input {
  padding: 2%;
  text-align: right;
  font-size: 0.5em;
}