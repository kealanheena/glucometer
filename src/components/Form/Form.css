label { 
  display: block;
  margin: auto;
  padding: 10%;
  font-size: var(--form-text-size);
  font-family: var(--display-font);
}

em {
  font-size: 0.4em;
}

input[type="number"] {
  font-size: var(--input-text-size);
  font-family: var(--display-font);
  width: 2.5em;
  padding: 0.2em;
  outline: none;
}

input[max="240"] {
  width: 3.5em;
}

input[type="submit"] {
  -moz-appearance: none;
  -webkit-appearance: none;
  appearance: none;
  outline: none;
 }

.form-container {
  text-align: center;  
  height: var(--form-height);
  width: var(--form-width);
}

.submit-button {
  color: var(--submit-button-color);
  color: var(--button-text-color);
  background-color: var(--display-background-color);
  font-family: var(--main-font);
  font-size: var(--button-text-size);
  font-weight: var(--button-text-weight);
  height: 10%;
  width: 60%;
  border-radius: 20px;
}

.submit-button:hover {
  color: var(--submit-button-hover-color);
  cursor: pointer;
}


.input-container {
  align-items: center;
  background: var(--display-background-color);
  height: var(--display-height);
  padding: 0 4%;
  margin: 4% 4% 0 4%;
  width: var(--display-width);
  border-top-left-radius: 15%;
  border-top-right-radius: 15%;
}

.logo {
  height: 100px;
}