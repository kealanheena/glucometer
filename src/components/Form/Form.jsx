import React from 'react';
// import PropTypes from 'prop-types';

class Form extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      carbs: 0,
      ratio: 0,
    }
  }

  carbsChangeHandler = (e) => {
    this.setState({ carbs: e.target.value });
  }

  ratioChangeHandler = (e) => {
    this.setState({ ratio: e.target.value });
  }

  render(){
    return (
      <form>
        <input 
          type="number"
          min='0'
          max='240'
          onChange={this.carbsChangeHandler}
        />
        <input
          type="number"
          min='0'
          max='55.4'
          onChange={this.ratioChangeHandler}
        />
        <input type="submit"/>
      </form>
    )}
}

export default Form;