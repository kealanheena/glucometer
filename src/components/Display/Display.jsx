import React from 'react';
import PropTypes from 'prop-types';
import './Display.css'

const Display = ({ displayValue }) => (
  <div className="display-container">
    <div className="display-value">
      { displayValue }
    </div>
  </div>
);

Display.propTypes = { displayValue: PropTypes.any.isRequired };

export default Display;
