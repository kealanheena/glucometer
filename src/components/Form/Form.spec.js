import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form'
import Display from '../Display/Display';

describe('Form', () => {
  let wrapper;

  beforeEach(() => wrapper = shallow(
    <Form
      submitHandler={jest.fn}
      carbs={''}
      ratio={''}
      carbsChangeHandler={jest.fn}
      ratioChangeHandler={jest.fn}
      displayValue={''} 
    />
  ));

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render a <form/>', () => {
    expect(wrapper.find('form').length).toEqual(1);
  });

  it('should render 3 <input/>\'s', () => {
    expect(wrapper.find('input').length).toEqual(3);
  });

  it('should render the Display Component', () => {
    expect(wrapper.containsMatchingElement(<Display displayValue={''} />)).toEqual(true);
  });
});
