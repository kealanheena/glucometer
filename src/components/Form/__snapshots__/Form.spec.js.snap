// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render correctly 1`] = `
<form
  className="form-container"
  onSubmit={[Function]}
>
  <div
    className="input-container"
  >
    <label>
      Carbs:
      <input
        max="240"
        min="0"
        onChange={[Function]}
        type="number"
      />
      <em>
        g
      </em>
    </label>
    <label>
      Ratio: 1:
      <input
        max="55.4"
        min="0"
        onChange={[Function]}
        type="number"
      />
      <em>
        mmol
      </em>
    </label>
  </div>
  <Display
    displayValue=""
  />
  <input
    className="submit-button"
    type="submit"
  />
  <img
    alt="glucometer-logo"
    className="logo"
    src="glucometer-logo.png"
  />
</form>
`;
